### ISSUE TITLE
Enforce Unique Tags

### ISSUE THREAD

**Issue by richardpeng** [alexgorbatchev/jquery-textext#85]
It would be great if the tags and autocomplete plugins could work together to enforce unique tags. You shouldn't be able to enter the same tag multiple times and a used tag should be removed from the autocomplete list.

**Comment by jesperavot** [comment]
You can achieve this quite easily using the 'isTagAllowed' event and an array.

``` javascript
var tagArray = [];
$('#textarea').textext({ plugins: 'tags' }).bind('isTagAllowed', function(e, data)
{
    if (in_array(data.tag.toLowerCase(), tagArray)) {
        data.result = false;
    } else {
        tagArray.push(data.tag.toLowerCase());
    }
});
```

An example in_array method can be found here: http://phpjs.org/functions/in_array/

Regarding the autocomplete list, you can filter that with the 'getSuggestions' event: http://textextjs.com/manual/examples/autocomplete-with-tags-and-filter.html

The only thing not mentioned is the deletion of tags. I've added a custom event for that and when a tag is removed, I remove it from the tagArray.

**Comment by ericorruption** [comment]
Actually, using both plugins makes 'isTagAllowed' trigger twice, which does not let the tagging happen. Having the same problem here.

**Comment by alexgorbatchev** [comment]
Same as #101, implemented in 2.0

**Comment by vishwa-** [comment]
to add to what @jesperavot has mentioned: This is one way to remove the element from the tagArray : 

Step 1: create a new custom function  to remove the element from the tag array: 

function customFunction(tag)
{
    var index = globalSkillArray.indexOf(tag);
    tagArray.splice(index, 1);  
}

Step 2: In your textext.js file modify the function p.removeTag = function(tag) as below

p.removeTag = function(tag)
{
    var self = this,
    core = self.core(),
     element
     ;
     if(tag instanceof $)
     {
          element = tag;
      tag = tag.data(CSS_TAG);
     }
     else
     {
          element = self.getTagElement(tag);
     }

```
 element.remove();
 self.updateFormCache();
 core.getFormData();
 core.invalidateBounds();

 customFunction(tag); //*****call your custom function here!*****
```

};
